src/bsk201909/secrets.md

```diff
@@ -0,0 +1,264 @@
+
+As soon as it finds a SealedSecret it uses the private key to decrypt the values and create a standard Kubernetes secret.
+
+You can verify that the secret was created successfully with:
+
+```terminal|command=1|title=bash
+kubectl get secrets mysecret -o yaml
+```
+
+Please notice that the secret created by the operator has the same name as the SealedSecret.
+
+You can use the Secrets in your Pods to inject environment variables or mount them as files.
+
+Since you can only decrypt the secrets with the private key (and that is safely stored in the cluster) you can sleep sweet dreams.
+
+Also, kubeseal supports secrets rotation.
+
+You can generate a new public and private key and re-encrypt your secrets.
+
+There are some downsides to consider, though.
+
+You can't see what's inside the secret.
+
+So every time you want to add a new value you might need to re-encrypt all values or create a separate secret.
+
+Sealed Secret use one key pair to encrypt all your secrets and keep it inside the cluster - without any protection (for example, using Hardware Security Model).
+
+There are two alternative tools to Sealed secrets that address those two shortcomings.
+
+The first is Helm Secrets.
+
+While the underlying mechanism to secure the secrets is similar to Sealed Secrets, there are some noteworthy differences.
+
+Helm secrets is capable of leveraging Helm to template secrets resources.
+
+If you work in a large team with several namespaces and you use Helm already, you might find Helm secrets more convenient than Sealed secrets.
+
+Helm secret has another advantage over Sealed Secrets - it's using the popular open source project SOPS (developed by Mozilla) for encrypting secrets.
+
+SOPS supports external key management systems, like AWS KMS, making it more secure as it's a lot harder to compromise the keys.
+
+With that said, both solutions share the same issues - to use them, you must have permissions to decrypt the secrets.
+
+If you work as part of a small team, perhaps this isn't a great deal.
+
+However, if you care about security and want to reduce your blast radius, you might not want to hand over the keys to your secrets to every DevOps and Developer in your team.
+
+Also, Helm Secrets is a Helm plugin and it is strongly coupled to Helm, making it harder to change to other templating mechanisms such as kustomize.
+
+You can learn more about Helm secrets on the official project page.
+
+The last solution I'll discuss is Kamus.
+
+> Full disclosure - the author is the lead developer.
+
+The architecture is similar to Sealed Secrets and Helm Secrets, however, Kamus lets you encrypt a secret for a specific application and only this application can decrypt it.
+
+The more granular permissions make Kamus more suitable to zero-trust environments with a high standard of security.
+
+Kamus works by associating a service account to your secrets.
+
+Only applications running with this service account are allowed to decrypt it.
+
```

Comment:
Explain how to install Kamus in the cluster and the Kamus CLI:

`````markdown
You can install Kamus in your cluster with the official Helm chart:

```terminal|command=1,2|title=bash
helm repo add soluto https://charts.soluto.io
helm upgrade --install kamus soluto/kamus
```

And you can install the Kamus CLI with:

```terminal|command=1|title=bash
npm install -g @soluto-asurion/kamus-cli
```
`````